### ISSUE TITLE
compilation fails when using async functions

### ISSUE THREAD

**Issue by dsaner** [RallyApps/rally-app-builder#83]
Promises are ugly.  Async functions are so much more readable.  When I try to use them however, I'm unable to compile my app.  Any chance we can get the compiler to work with them?

Thanks!
Dustin

**Comment by krmorse** [comment]
This is a combination of using an old version of the uglify package and needing to update your .jshintrc in your app to target esversion 6 instead of 5.  I created a pr to start playing with upgrading uglify.  Will keep you posted.

Note that since rally app builder currently does not do any transpilation async/await and other fancy es6 syntax like () => {} will only work in newer browsers.  Rally still technically supports ie11, which is partly why we haven't updated the app builder to be able to take advantage of the newest stuff.

But I agree with you, it would be nice to use some of it if you're in an environment where all your users are on latest Chrome or similar.

**Comment by krmorse** [comment]
There's also a larger issue around this with jshint: https://github.com/jshint/jshint/issues/2604

The right path forward is probably to replace jshint with eslint altogether, but that's a bigger change than is reasonable right now I think.

How do you feel about surrounding your async/await lines with jshint disable comments?  Is that a showstopper @dsaner?

**Comment by dsaner** [comment]
That would be fine by me as a workaround for now.  Is there any other setup I need to do besides upping my esversion to 6?  I tried adding some jshint ignore:start/end and jshint ignore:line comments, but they don't seem to be helping.

**Comment by ferentchak** [comment]
@dsaner  I go through and replace all the checks in the file to false from true.

One really big gotcha that you should know is that the Deft promises predate the ES6 promise spec and do not have the same signature. You will need to wrap them if you want the error handling in your await to function how you expect.

```
    getPromiseFromDeft(deftPromise) {
        if (!deftPromise.$className) {
            return deftPromise;
        }
        const promise = Cisco.getPromise();
        deftPromise.then({
            success: (...args) => {
                promise.resolve(...args);
            },
            failure: (...args) => {
                promise.reject(...args);
            }
        });
        return promise;
    }
```

**Comment by dsaner** [comment]
Yeah!  jshint ignore comments surrounding my async/await stuff and I've got a clean compile and functioning code.  Thanks!
